File: backend/app/schemas/storage.py
from datetime import datetime

from pydantic import BaseModel


class PVCInfo(BaseModel):
    name: str
    namespace: str
    status: str
    capacity: str | None
    storage_class: str | None
    access_modes: list[str]
    volume_name: str | None
    created_at: datetime | None


class PVInfo(BaseModel):
    name: str
    status: str
    capacity: str | None
    access_modes: list[str]
    storage_class: str | None
    reclaim_policy: str | None
    claim_namespace: str | None
    claim_name: str | None
    created_at: datetime | None


class PVCCreate(BaseModel):
    name: str
    namespace: str
    storage_class: str
    access_modes: list[str] = ["ReadWriteOnce"]
    size: str


class StorageClassInfo(BaseModel):
    name: str
    provisioner: str
    reclaim_policy: str
    binding_mode: str
    is_default: bool
    created_at: datetime | None

File: frontend/src/types/storage.ts
export interface PVCInfo {
  name: string;
  namespace: string;
  status: string;
  capacity: string | null;
  storage_class: string | null;
  access_modes: string[];
  volume_name: string | null;
  created_at: string | null;
}

export interface PVInfo {
  name: string;
  status: string;
  capacity: string | null;
  access_modes: string[];
  storage_class: string | null;
  reclaim_policy: string | null;
  claim_namespace: string | null;
  claim_name: string | null;
  created_at: string | null;
}

export interface PVCCreate {
  name: string;
  namespace: string;
  storage_class: string;
  access_modes: string[];
  size: string;
}

export interface StorageClassInfo {
  name: string;
  provisioner: string;
  reclaim_policy: string;
  binding_mode: string;
  is_default: boolean;
  created_at: string | null;
}
